### DISCUSSION TITLE
[Feature request] Extend Prisma client model types to include client extension properties

### DISCUSSION THREAD

**Discussion by ludwigbacklund** [prisma/prisma#19106] [Q&A]
## Problem

I add a __typename property to all my prisma model objects via a client extension (https://github.com/prisma/prisma/issues/5315#issuecomment-1447116621, but with __typename instead of $kind). However, when I import say the User model type, it does not have this __typename property in the type.

I want to be able to extend the User model type to know that it now has a __typename property on it. I suspect that these model types that I'm importing are describing the database model structure of a User, and not what I get from my extended prisma client. Is there perhaps a different way I can reference the User model type, but as it would look coming out of my extended prisma client?

## Suggested solution

One way I could see this working is being able to do something like ExtendedPrismaClient['User'], but nothing like that exists on the extended prisma client type today.

(https://github.com/prisma/prisma/discussions/18460)

**Comment by janpio** [comment]
Sorry for missing that this was a discussion before :/

**Reply by janpio** [reply]
Now again an issue at https://github.com/prisma/prisma/issues/19112
